core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java

```diff
@@ -81,11 +86,16 @@ public void shutdown() {}
     @Override
     public void handleResolvedAddressGroups(
         List<EquivalentAddressGroup> servers, Attributes attributes) {
+      if (attributes.get(ATTR_LOAD_BALANCING_CONFIG) != null) {
+        throw new IllegalArgumentException(
+            "Unexpected ATTR_LOAD_BALANCING_CONFIG from upstream: "
+            + attributes.get(ATTR_LOAD_BALANCING_CONFIG));
+      }
       Map<String, Object> configMap = attributes.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
-      LoadBalancerProvider newlbp;
+      PolicySelection selection;
       try {
-        newlbp = decideLoadBalancerProvider(servers, configMap);
-      } catch (PolicyNotFoundException e) {
+        selection = decideLoadBalancerProvider(servers, configMap, helper.getChannelLogger());
+      } catch (PolicyException e) {
         Status s = Status.INTERNAL.withDescription(e.getMessage());
```

Comment:
chain the cause?

Reply:
I intentionally do not attach the exception to the status because the PolicyException's message has all information.